samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -267,8 +267,8 @@ mfxStatus CDecodingPipeline::GetImpl(const sInputParams & params, mfxIMPL & impl
 
     // If d3d11 surfaces are used ask the library to run acceleration through D3D11
     // feature may be unsupported due to OS or MSDK API version
-
-    if (D3D11_MEMORY == params.memType || D3D11_MEMORY == m_memType)
+    // prefer D3D11 by default
+    if (D3D9_MEMORY != params.memType || D3D9_MEMORY != m_memType)
```

Comment:
Please check how it behaves on Linux

Reply:
Placed it under `D3D_SURFACES_SUPPORT` define.